### ISSUE TITLE
Consider making `click` an optional dependency

### ISSUE THREAD

**Issue by althonos** [archspec/archspec#108]
Hi !

This is a very neat package, and it addresses a very common problem I have to handle in several packages I'm developing: CPU feature detection at runtime. So, I'm considering switching to `archspec` instead of manually building and wrapping `libcpu_features` in every different package.

Now, the only (small) blocker is that I wouldn't be needing the `archspec` CLI, so for distribution I'm a bit reluctant to add `click` as a transitive dependency when it's not being used at all in a library package. However, since it's only ever required by the CLI, would you agree to make it an optional dependency ?

This is quite easy with Poetry, you can make `click` optional:
```toml
[tool.poetry.dependencies]
python = "3.6.15 || ^3.7"
click = { version = "^8", optional = true }
```
Then declare an *extra* feature group which depends on `click`:
```toml
[tool.poetry.extras]
cli = ["click"]
```
Then you can define the `archspec` CLI as requiring the `cli` extra, which will make `click` mandatory, with:
```toml
[tool.poetry.scripts]
archspec = { callable = "archspec.cli:main", extras = ["cli"] }
```
This way, end-users can specificy whether they need the `archspec` CLI or just the `archspec` module.

Please tell me what you think! Cheers.

**Comment by alalazo** [comment]
That would be absolutely acceptable from my point of view. Do you have time / will to submit a PR for that?

**Comment by althonos** [comment]
Sure! I'll try to write one tomorrow.

**Comment by tgamblin** [comment]
Honestly I think it would be good to switch to argparse anyway, to make archspec consistent with `spack` (and to remove the dependency) -- we could do that too, but making it optional first is fine with me.

**Comment by althonos** [comment]
I'm also happy to write an `argparse` implementation, the current CLI is simple enough! 😇

**Comment by tgamblin** [comment]
that would be great!

**Comment by tgamblin** [comment]
Closed by #109
